pkg/controller/route/routecontroller.go

```diff
@@ -23,9 +23,19 @@ import (
 
 	"github.com/golang/glog"
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
 	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	workerGoroutines = 5
```

Comment:
Any reason for 5 workers?

If we assume N nodes, and 5 seconds for a route insert, we're saying that route initialization will take at least N seconds. Insignificant for small clusters, but might actually be the bottleneck on larger clusters.